invenio_records_resources/records/api.py

```diff
@@ -131,42 +132,42 @@ def __init__(self, object_model=None, file_model=None):
         self.file_model = file_model or object_model.file
 
     @classmethod
-    def from_dict(cls, data, bucket):
+    def from_dump(cls, data):
         """Construct a file wrapper from a dictionary."""
         file_args = dict(
             id=data["file_id"],
-            storage_class=data.get("storage_class"),
             size=data.get("size"),
             checksum=data.get("checksum"),
         )
-        if "uri" in data:
-            file_args["uri"] = data["uri"]
-
         fi = FileInstance(**file_args)
         obj = ObjectVersion(
-            version_id=data["version_id"],
+            version_id=data["object_version_id"],
             key=data["key"],
             file_id=data["file_id"],
+            file=fi,
             _mimetype=data["mimetype"],
             is_head=True,
-            bucket=bucket,
-            bucket_id=data.get("bucket_id", bucket.id),
+            bucket_id=data["bucket_id"],
         )
         return cls(object_model=obj, file_model=fi)
 
     def dumps(self):
         """Dump file model attributes of the object."""
-        return {
-            "version_id": str(self.object_model.version_id),
-            "key": self.object_model.key,
-            "bucket_id": str(self.object_model.bucket_id),
```

Comment:
The bucket id is not needed here because there's an optimisation in the FilesManager/PartialFileDumper. 

The record data model has a separate field for the bucket id which is already dumped into the Elasticsearch index for the record, hence I did not want to duplicate the information for every single file in the record. Therefore the ``FilesManager`` when loading data from the Elasticsearch index will automatically inject record id and bucket id so that the loading in ``from_dump()`` has it available. 

See https://github.com/inveniosoftware/invenio-records-resources/pull/463/files#diff-88e8e04ee3963cbc7f0779e53281e2478b0ea0d18cc97195532387f5c3a2fd5eR201